### PR TITLE
Make --cluster-name a required CLI argument

### LINKED ISSUE #820: Cluster with name prefix ocs* already exists when not passing --cluster-name
I am attempting to execute a deployment without passing `--cluster-name` and therefore using the default constructed `cluster_name`. I am seeing the below Exception being raised, apparently due to the prefix `ocs*` already being used.

By default `cluster_name = f"ocs-ci-{getuser()[:8]}"`. This entire string should be used to find non-terminated instances and not just `ocs`. As it currently works, if there is a cluster up and running in AWS that used the default `cluster_name`, no one will be able to execute a deployment using the default `cluster_name`. 

```
self = <ocs_ci.deployment.aws.AWSIPI object at 0x10b6ea908>, log_cli_level = 'INFO'

    def deploy_ocp(self, log_cli_level='DEBUG'):
        """
        Deployment specific to OCP cluster on this platform

        Args:
            log_cli_level (str): openshift installer's log level
                (default: "DEBUG")
        """
        if not config.DEPLOYMENT.get('force_deploy_multiple_clusters'):
            cluster_name = config.ENV_DATA['cluster_name']
            prefix = cluster_name.split("-")[0] + '*'
            if self.check_cluster_existence(prefix):
                raise SameNamePrefixClusterAlreadyExistsException(
>                   f"Cluster with name prefix {prefix} already exists. "
                    f"Please destroy the existing cluster for a new cluster deployment"
                )
E               ocs_ci.ocs.exceptions.SameNamePrefixClusterAlreadyExistsException: Cluster with name prefix ocs* already exists. Please destroy the existing cluster for a new cluster deployment

ocs_ci/deployment/aws.py:151: SameNamePrefixClusterAlreadyExistsException
```

### PR DESCRIPTION
**Pull request by gklein**
Resolve #820  by making --cluster-name mandatory  

Signed-off-by: Gil Klein <gklein@users.noreply.github.com>

### CODE REVIEW

**Comment by vasukulkarni** [comment]
Can you please make this mandatory only for --deployment(mutual dependency), The --cluster-name is not required for running tests.

**Comment by gklein** [comment]
> Can you please make this mandatory only for --deployment(mutual dependency), The --cluster-name is not required for running tests.

Right, I guess we will need it for both --deployment and --teardown

**Comment by vasukulkarni** [comment]
> > Can you please make this mandatory only for --deployment(mutual dependency), The --cluster-name is not required for running tests.
> 
> Right, I guess we will need it for both --deployment and --teardown

I remember not using --cluster-name for --teardown , it needs --cluster-path.

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 1]
```diff
@@ -207,14 +209,17 @@ def process_cluster_cli_params(config):
...
-    if not cluster_name:
-        cluster_name = f"ocs-ci-{getuser()[:8]}"
     ocsci_config.RUN['cli_params']['teardown'] = get_cli_param(config, "teardown", default=False)
     ocsci_config.RUN['cli_params']['deploy'] = get_cli_param(config, "deploy", default=False)
     ocsci_config.RUN['cli_params']['io_in_bg'] = get_cli_param(config, "io_in_bg", default=False)
     ocsci_config.ENV_DATA['cluster_name'] = cluster_name
     ocsci_config.ENV_DATA['cluster_path'] = cluster_path
     get_cli_param(config, 'collect-logs')
+    if (
+        ocsci_config.RUN["cli_params"]["deploy"]
```
Comment (ebenahar) [inline ocs_ci/framework/pytest_customization/ocscilib.py]: Replace with:
```python
ocsci_config.RUN.get("cli_params").get("deploy")
```

Please apply to all

File: ocs_ci/framework/pytest_customization/tests/test_pytest.py [hunk 2]
```diff
@@ -84,7 +84,11 @@ def test_demo_parametrized_config(item):
             - 1
             - 2
         """))
-    pytest_arguments = ['-v', f'--ocsci-conf={conf_file}', f'--cluster-path={tmpdir}']
+    pytest_arguments = ['-v',
```
Comment (petr-balogh) [inline ocs_ci/framework/pytest_customization/tests/test_pytest.py]: bracket style
```
[
    contetn1,
    content2,
]
```

**Comment by vasukulkarni** [comment]
I am removing needs testing flag, since you added verified flag.

**Pull request merged**